File: experiments-local/experiment-15/runs/agent=omp_language=java_model=openrouter/z-ai/glm-5.2_tooling=none/rep3/src/main/java/com/example/bookapi/model/Book.java
package com.example.bookapi.model;

/**
 * A book in the collection.
 *
 * @param id     database-generated identifier
 * @param title  non-blank title
 * @param author non-blank author
 * @param year   publication year (nullable)
 * @param isbn   ISBN identifier (nullable)
 */
public record Book(Long id, String title, String author, Integer year, String isbn) {
}

File: experiments-local/experiment-mu-primeagent-easy/runs/agent=prime_language=go_model=openrouter/z-ai/glm-5.2_tooling=none/rep1-failed/model/book.go
package model

// Book represents a single book in the collection.
type Book struct {
	ID     int    `json:"id"`
	Title  string `json:"title"`
	Author string `json:"author"`
	Year   int    `json:"year"`
	ISBN   string `json:"isbn"`
}

// BookInput is the payload used when creating or updating a book.
type BookInput struct {
	Title  string `json:"title"`
	Author string `json:"author"`
	Year   int    `json:"year"`
	ISBN   string `json:"isbn"`
}

// Validate checks that required fields are present. It returns a slice of
// human-readable error messages for every invalid field.
func (b BookInput) Validate() []string {
	var errs []string
	if b.Title == "" {
		errs = append(errs, "title is required")
	}
	if b.Author == "" {
		errs = append(errs, "author is required")
	}
	if b.Year < 0 {
		errs = append(errs, "year must be a non-negative integer")
	}
	return errs
}
